src/Draco.Fuzzer/Testing/Generators/RandomValidTokenGenerator.cs

```diff
@@ -0,0 +1,114 @@
+
+    private IEnumerable<SyntaxToken> LexTokens(Lexer lexer)
+    {
+        var tokens = new List<SyntaxToken>();
+        while (true)
+        {
+            var tok = lexer.Lex();
+            tokens.Add(tok);
+            if (tok.Kind == TokenKind.EndOfInput) break;
+        }
+        return tokens;
+    }
+
+    private string GetRandomToken()
+    {
+        var max = Enum.GetValues(typeof(TokenKind)).Length;
+        var rand = this.random.Next(max);
+        var text = this.GenerateTokenText((TokenKind)rand);
+        var toAppend = this.random.Next(2) == 0 ? "" : " "; //Append space or not
+        return $"{text}{toAppend}";
+    }
+
+    private string GenerateTokenText(TokenKind kind) => kind switch
+    {
+        TokenKind.Identifier => this.GenerateValidIdentifier(),
+        TokenKind.LiteralInteger => this.GenerateInteger(),
+        TokenKind.LiteralFloat => this.GenerateFloat(),
+        TokenKind.LiteralCharacter => this.GenerateChar(),
+        TokenKind.LineStringStart => this.GenerateLineStringStart(),
+        TokenKind.LineStringEnd => this.GenerateLineStringEnd(),
+        TokenKind.MultiLineStringStart => this.GenerateMultiLineStringStart(),
+        TokenKind.MultiLineStringEnd => this.GenerateMultiLineStringEnd(),
+        TokenKind.StringContent => this.GenerateStringContent(),
+        TokenKind.StringNewline => Environment.NewLine,
+        TokenKind.InterpolationStart => this.GenerateInterpolationStart(),
+        _ => SyntaxFacts.GetTokenText(kind)!
+    };
+
+    private string GenerateValidIdentifier()
+    {
+        var lowerBound = (int)'a';
+        var upperBound = ((int)'z' + 1);
+        var builder = new StringBuilder();
+        builder.Append((char)this.random.Next(lowerBound, upperBound)); // Valid alpha char
+        var len = this.random.Next(25);
+        for (var i = 0; i < len; i++)
+        {
+            builder.Append((char)this.random.Next(lowerBound, upperBound));
+        }
+        return builder.ToString();
+    }
+
+    private string GenerateInteger() => this.random.Next().ToString();
+
+    private string GenerateFloat() => this.random.NextDouble().ToString();
+
+    private string GenerateChar() => $"'{this.charset[this.random.Next(this.charset.Length)]}'";
+
+    private string GenerateDelimiters() => new string('#', this.random.Next(4));
+
+    private string GenerateLineStringStart() => $"{this.GenerateDelimiters()}\"";
+
+    private string GenerateLineStringEnd() => $"\"{this.GenerateDelimiters()}";
+
+    private string GenerateMultiLineStringStart() => $"{this.GenerateDelimiters()}\"\"\"";
+
+    private string GenerateMultiLineStringEnd() => $"\"\"\"{this.GenerateDelimiters()}";
+
+    private string GenerateStringContent() => new string(Enumerable.Range(0, this.random.Next(128)).Select(x => this.charset[this.random.Next(this.charset.Length)]).ToArray());
+
+    private string GenerateInterpolationStart() => $"\\{this.GenerateDelimiters()}{{";
```

Comment:
Nice!

Reply:
We could use some utility functions eventually, but as a start this is totally fine